lib/rdf/transaction.rb

```diff
@@ -81,38 +119,48 @@ def context=(value)
     # @param  [Hash{Symbol => Object}]  options
     # @option options [Boolean]         :mutable (false)
     #    Whether this is a read-only or read/write transaction.
-    # @option options [RDF::Resource]   :graph_name (nil)
-    #   Name of named graph to be affected if `inserts` or `deletes`
-    #   do not have a `graph_name`.
-    # @option options [RDF::Resource]   :graph  (nil)
-    #   Alias for `:graph_name`.
-    # @option options [RDF::Enumerable] :insert (RDF::Graph.new)
-    # @option options [RDF::Enumerable] :delete (RDF::Graph.new)
+    # @option options [Queryable]       :snapshot
+    #    A queryable snapshot of the repository for isolated reads. Defaults to
+    #    `#repository` as an unisolated query target.
     # @yield  [tx]
     # @yieldparam [RDF::Transaction] tx
-    def initialize(options = {}, &block)
-      @options = options.dup
-      if @options.has_key?(:context)
-        raise ArgumentError, "The :contexts option to Mutable#load is deprecated in RDF.rb 2.0, use :graph_name instead. Called from #{Gem.location_of_caller.join(':')}" if RDF::VERSION.to_s >= '2.0'
-        warn "[DEPRECATION] the :contexts option to Mutable#load is deprecated in RDF.rb 2.0, use :graph_name instead. Called from #{Gem.location_of_caller.join(':')}"
-        @options[:graph_name] ||= @options.delete(:context)
-      end
-      @mutable = @options.delete(:mutable) || false
-      @graph_name = @options.delete(:graph) || @options.delete(:graph_name)
-      @inserts = @options.delete(:insert)   || []
-      @deletes = @options.delete(:delete)   || []
-      @inserts.extend(RDF::Enumerable) unless @inserts.kind_of?(RDF::Enumerable)
-      @deletes.extend(RDF::Enumerable) unless @deletes.kind_of?(RDF::Enumerable)
-
+    def initialize(repository, options = {}, &block)
+      @repository = repository
+      @options  = options.dup
+      @snapshot = @options.delete(:snapshot) { @repository }
```

Comment:
This interface just came together in the last hour. I'm somewhat uncertain about it, and open to suggestions for revision.